Content.Server/RandomAppearance/RandomAppearanceComponent.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Robust.Shared.Analyzers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Reflection;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.RandomAppearance;
+
+[RegisterComponent, ComponentProtoName("RandomAppearance")]
+[Friend(typeof(RandomAppearanceSystem))]
+public class RandomAppearanceComponent : Component, ISerializationHooks
+{
+    [DataField("spriteStates")]
+    public string[] SpriteStates = {"0", "1", "2", "3", "4"};
+
+    /// <summary>
+    ///     What appearance enum key should be set to the random sprite state?
+    /// </summary>
+    [DataField("key", required: true)]
+    public string EnumKeyRaw = default!;
+
+    /// <summary>
+    ///     The actual enum after reflection.
+    /// </summary>
+    public object? EnumKey;
```

Comment:
TryParseEnumReference returns an enum reference. Why not just use Enum instead of object, then you don't have to do the cast in the system?